### DISCUSSION TITLE
Set system tray menu items to disabled at start up

### DISCUSSION THREAD

**Discussion by tinomatic** [tauri-apps/tauri#2200] [Q&A] [answered]
Hi,

I have a question open on [stackoverflow](https://stackoverflow.com/questions/68322880/set-system-tray-menu-items-to-disabled-when-starting-tauri-tray-app) regarding disabling (some) tray menu items at app start up. Basically, when I try to access the tray items in .setup I'm told the tray is not configured:

`thread 'main' panicked at 'tray not configured; use the Builder#system_tray API first.'`

That makes sense, but then where can I set the state of the items? I did try to set state when creating the menu items themselves before adding them to the tray menu, but there doesn't seem to be a method to do that.

My codes looks like this:

```rs
let opt1 = CustomMenuItem::new("opt1".to_string(), "Option 1");
let opt2 = CustomMenuItem::new("opt2".to_string(), "Option 2");

let tray_menu = SystemTrayMenu::new()
    .add_item(opt1)
    .add_item(opt2);

tauri::Builder::default()
    .system_tray(SystemTray::new().with_menu(tray_menu))

.setup(|app| {
    ...

    let tray_handle = app.handle().tray_handle();
    tray_handle.get_item(&"opt1".to_string()).set_enabled(false);

    ...
})
.run(tauri::generate_context!())
.expect("error while running tauri application");
```

TIA

**Comment by woubuc** [comment, marked as answer]
You can use `.disabled()` on your `CustomMenuItem` to set it to disabled right away, before you add it to the menu.

From my code:
```rs
SystemTray::new().with_menu(SystemTrayMenu::new()
    .add_item(CustomMenuItem::new("action".to_string(), "Loading...").disabled())
    .add_item(CustomMenuItem::new("quit".to_string(), "Quit"))
)
```

This results in a disabled menu item that says "Loading...":
![image](https://user-images.githubusercontent.com/1015540/126667179-00ce59e1-37ba-46fe-953f-6386b2cb7b48.png)

**Reply by tinomatic** [reply]
Thanks! That is what I needed. Will post follow-up question related to later re-enabling them in a new thread.
